base/common/src/main/java/org/dogtagpki/acme/ACMEAlgorithm.java

```diff
@@ -0,0 +1,44 @@
+package org.dogtagpki.acme;
+
+/**
+ * @author Minsu Park
+ */
+
+public enum ACMEAlgorithm {
+    // RFC 7518 Appendix A.1
+    // Digital Signature/MAC Algorithm Identifier Cross-Reference
+    HS256("HS256", "HmacSHA256"),
+    HS384("HS384", "HmacSHA384"),
+    HS512("HS512", "HmacSHA512"),
+    RS256("RS256", "SHA256withRSA"),
+    RS384("RS384", "SHA384withRSA"),
+    RS512("RS512", "SHA512withRSA"),
+    ES256("ES256", "SHA256withECDSA"),
+    ES384("ES384", "SHA384withECDSA"),
+    ES512("ES512", "SHA512withECDSA"),
+    PS256("PS256", "SHA256withRSAandMGF1"),
+    PS384("PS384", "SHA384withRSAandMGF1"),
+    PS512("PS512", "SHA512withRSAandMGF1");
+
+    private String alg;
+    private String jca;
+
+    private ACMEAlgorithm(String alg, String jca) {
+        this.alg = alg;
+        this.jca = jca;
+    }
+
+    public String getJCA() {
+        return jca;
+    }
+
+    public static ACMEAlgorithm fromString(String alg) throws Exception {
+        for (ACMEAlgorithm a : ACMEAlgorithm.values()) {
+            if (a.alg == alg) {
```

Comment:
String comparison needs to be done using `equals()` or `equalsIgnoreCase()`.

Reply:
Thanks for the review! Rather than Implementing this myself, it seems I should have been using the [valueOf](https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#valueOf-java.lang.Class-java.lang.String-) function that is a part of Enum objects. I think this paired with .toUpperCase() should do the trick.